File: wrappers/dotnet/src/BbsSignatures/Models/SignRequest.cs
namespace BbsSignatures
{
    /// <summary>
    /// A request to create a BBS signature for a set of messages from a BLS12-381 key pair
    /// </summary>
    public class SignRequest
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SignRequest"/> class.
        /// </summary>
        /// <param name="keyPair">BLS12-381 key pair</param>
        /// <param name="messages">Messages to sign</param>
        public SignRequest(BlsKeyPair keyPair, string[] messages)
        {
            KeyPair = keyPair;
            Messages = messages;
        }

        /// <summary>
        /// BLS12-381 key pair
        /// </summary>
        /// <value>
        /// The key pair.
        /// </value>
        public BlsKeyPair KeyPair { get; set; }

        /// <summary>
        /// Messages to sign
        /// </summary>
        /// <value>
        /// The messages.
        /// </value>
        public string[] Messages { get; set; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/SignRequest.py
from typing import List
from .keys import BlsKeyPair


class SignRequest:
    def __init__(self, key_pair: BlsKeyPair, messages: List[str]) -> None:
        """ A request to create a BBS signature for a set of messages from a BLS12-381 key pair
        Args:
            key_pair: BLS12-381 key pair
            messages: Messages to sign
        """
        self.key_pair = key_pair
        self.messages = messages
